tests/TestCase/Db/Adapter/PostgresAdapterTest.php

```diff
@@ -132,6 +132,36 @@ public function testQuoteSchemaName()
         $this->assertEquals('"schema.schema"', $this->adapter->quoteSchemaName('schema.schema'));
     }
 
+    public function testGetGlobalSchemaName()
+    {
+        $config = ConnectionManager::getConfig('test');
+        $config['schema'] = 'test_schema';
+        ConnectionManager::drop('test');
+        ConnectionManager::setConfig('test', $config);
+        // Emulate the results of Util::parseDsn()
+        $this->config = [
+            'adapter' => 'postgres',
+            'connection' => ConnectionManager::get('test'),
+            'database' => $config['database'],
+        ];
+
+        $this->adapter = new PostgresAdapter($this->config, $this->getConsoleIo());
+
+        $this->adapter->dropAllSchemas();
+        $this->adapter->createSchema('test_schema');
+
+        $this->adapter->disconnect();
+
+        $this->assertEquals('"test_schema"."table"', $this->adapter->quoteTableName('table'));
+
+        $config = ConnectionManager::getConfig('test');
+        unset($config['schema']);
+        ConnectionManager::drop('test');
+        ConnectionManager::setConfig('test', $config);
+
+        $this->adapter->disconnect();
```

Comment:
I couldn't really think of a better way to test this besides this method. Since setting the schema does make the rest of the tests fail.

Reply:
Perhaps don't drop the other schema? or overwrite the `test` connection?  Instead, you could create a new connection, to the same database as test, and a new schema. If you then create a table in a new schema, you should be able to find it in `$connection->listTables()` on the new schema but not the old. Then you wouldn't modify the existing configuration that other tests are using.